bitseq/sequence.go

```diff
@@ -308,17 +308,17 @@ func (h *Handle) set(ordinal, start, end uint64, any bool, release bool) (uint64
 
 	for {
 		var store datastore.DataStore
-		h.Lock()
+		h.RLock()
 		store = h.store
-		h.Unlock()
+		h.RUnlock()
 		if store != nil {
-			if err := store.GetObject(datastore.Key(h.Key()...), h); err != nil && err != datastore.ErrKeyNotFound {
+			if err = store.GetObject(datastore.Key(h.Key()...), h); err != nil && err != datastore.ErrKeyNotFound {
 				return ret, err
 			}
 		}
 
-		h.Lock()
 		// Get position if available
```

Comment:
@fcrisciani thats correct. But in this case, the `set` method already handles this scenario by retrying the entire operation if the AtomicPut fails in the store. Also the Store's AtomicPut guarantees the write consistency with the use of Object Index. If an object that is writing to the store has an older data Index, then the write will fail and the `set` will retry the operation. Please look the logic below in this method, where we handle the `types.RetryError` and redo the entire `set` again.